## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 511210`
Commit: `c61168cd46471ab7dd681344e90d7e4e4f0c25a6`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.exa-py</li>
    <li>python314Packages.exa-py.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python313Packages.exa-py</li>
    <li>python313Packages.exa-py.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.exa-py</li>
    <li>python314Packages.exa-py.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python313Packages.exa-py</li>
    <li>python313Packages.exa-py.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.exa-py</li>
    <li>python314Packages.exa-py.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python313Packages.exa-py</li>
    <li>python313Packages.exa-py.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.exa-py</li>
    <li>python314Packages.exa-py.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python313Packages.exa-py</li>
    <li>python313Packages.exa-py.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python314Packages.exa-py</summary>
<pre>    @pytest.mark.timeout(30)

tests/integration/test_search_integration.py:53
  /build/source/tests/integration/test_search_integration.py:53: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(15)

tests/integration/test_search_integration.py:66
  /build/source/tests/integration/test_search_integration.py:66: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(20)

tests/integration/test_search_integration.py:124
  /build/source/tests/integration/test_search_integration.py:124: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(15)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_list_all_single_page - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_list_all_multiple_pages - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_get_all_returns_list - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
============ 3 failed, 171 passed, 35 skipped, 7 warnings in 6.65s =============</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python314Packages.exa-py</summary>
<pre>    @pytest.mark.timeout(30)

tests/integration/test_search_integration.py:53
  /build/source/tests/integration/test_search_integration.py:53: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(15)

tests/integration/test_search_integration.py:66
  /build/source/tests/integration/test_search_integration.py:66: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(20)

tests/integration/test_search_integration.py:124
  /build/source/tests/integration/test_search_integration.py:124: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(15)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_list_all_single_page - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_list_all_multiple_pages - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_get_all_returns_list - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
============ 3 failed, 171 passed, 35 skipped, 7 warnings in 4.41s =============</pre>
</details>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python314Packages.exa-py</summary>
<pre>    @pytest.mark.timeout(30)

tests/integration/test_search_integration.py:53
  /nix/var/nix/builds/nix-45183-1957100051/source/tests/integration/test_search_integration.py:53: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(15)

tests/integration/test_search_integration.py:66
  /nix/var/nix/builds/nix-45183-1957100051/source/tests/integration/test_search_integration.py:66: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(20)

tests/integration/test_search_integration.py:124
  /nix/var/nix/builds/nix-45183-1957100051/source/tests/integration/test_search_integration.py:124: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(15)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_list_all_single_page - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_list_all_multiple_pages - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
FAILED tests/unit/test_search_monitors.py::TestAsyncSearchMonitorsListAll::test_get_all_returns_list - RuntimeError: There is no current event loop in thread &#x27;MainThread&#x27;.
============ 3 failed, 171 passed, 35 skipped, 7 warnings in 3.04s =============</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python314Packages.exa-py</summary>
<pre>tests/unit/test_pydantic_schemas.py::TestSchemaConversion::test_nested_pydantic_model_conversion PASSED [ 23%]
tests/unit/test_pydantic_schemas.py::TestSchemaConversion::test_dict_passthrough PASSED [ 24%]
tests/unit/test_pydantic_schemas.py::TestSchemaConversion::test_legacy_jsonschema_support PASSED [ 24%]
tests/unit/test_pydantic_schemas.py::TestSchemaConversion::test_invalid_type_raises_error PASSED [ 25%]
tests/unit/test_pydantic_schemas.py::TestTypeAnnotations::test_jsonschema_input_accepts_pydantic PASSED [ 25%]
tests/unit/test_pydantic_schemas.py::TestTypeAnnotations::test_backward_compatibility_with_typeddict PASSED [ 26%]
tests/unit/test_pydantic_schemas.py::TestIntegrationMocking::test_summary_with_pydantic_model PASSED [ 26%]
tests/unit/test_pydantic_schemas.py::TestIntegrationMocking::test_answer_with_pydantic_model PASSED [ 27%]
tests/unit/test_search_api.py::test_contentstatus_parsing_offline PASSED [ 27%]
tests/unit/test_search_api.py::test_answerresponse_accepts_dict PASSED   [ 28%]
tests/unit/test_search_api.py::test_search_accepts_user_location_offline PASSED [ 28%]
tests/unit/test_search_api.py::test_search_accepts_additional_queries_offline PASSED [ 29%]
tests/unit/test_search_api.py::test_search_accepts_instant_type_offline PASSED [ 29%]
tests/unit/test_search_api.py::test_search_accepts_deep_reasoning_params_offline PASSED [ 30%]
tests/unit/test_search_api.py::test_search_accepts_output_schema_on_fast_search_offline PASSED [ 30%]
tests/unit/test_search_api.py::test_search_accepts_system_prompt_on_auto_search_offline PASSED [ 31%]
tests/unit/test_search_api.py::test_search_accepts_deep_lite_additional_queries_offline PASSED [ 31%]
tests/unit/test_search_api.py::test_search_rejects_stream_flag_offline PASSED [ 32%]
tests/unit/test_search_api.py::test_stream_search_streams_openai_style_chunks_offline PASSED [ 32%]
tests/unit/test_search_api.py::test_</pre>
</details>
</details>

--- a/pkgs/development/python-modules/exa-py/default.nix
+++ b/pkgs/development/python-modules/exa-py/default.nix
@@ -1,0 +1,85 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build
+  poetry-core,
+
+  # deps
+  httpcore,
+  httpx,
+  openai,
+  pydantic,
+  python-dotenv,
+  requests,
+  typing-extensions,
+
+  # tests
+  pytest-asyncio,
+  pytest-cov-stub,
+  pytest-mock,
+  pytestCheckHook,
+
+  # passthru
+  unstableGitUpdater,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "exa-py";
+  version = "2.12.0-unstable-2026-04-15";
+  pyproject = true;
+  __structuredAttrs = true;
+
+  # pypi doesn't include tests but there aren't any upstream git tags
+  src = fetchFromGitHub {
+    owner = "exa-labs";
+    repo = "exa-py";
+    rev = "af7f88999763f1cb7e3c4a67f4aa24cef5f6eb11";
+    hash = "sha256-7rLEvjngSRObFdT1DcrZfqWBCsvWVxdNIgxBNhNNk+4=";
+  };
+
+  # https://github.com/pytest-dev/pytest-asyncio/issues/658
+  # default behaviour changes with new python version.
+  # planning on trying to vendor upstream
+  postPatch = ''
+    substituteInPlace tests/unit/test_search_monitors.py --replace-fail \
+      'get_event_loop().run_until_complete' 'run'
+  '';
+
+  build-system = [
+    poetry-core
+  ];
+
+  dependencies = [
+    httpcore
+    httpx
+    openai
+    pydantic
+    python-dotenv
+    requests
+    typing-extensions
+  ];
+
+  pythonImportsCheck = [
+    "exa_py"
+  ];
+
+  nativeCheckInputs = [
+    pytest-asyncio
+    pytest-cov-stub
+    pytest-mock
+    pytestCheckHook
+  ];
+
+  pytestFlags = [ "tests/" ];
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = {
+    description = "Official Python SDK for Exa, the web search API for AI";
+    homepage = "https://github.com/exa-labs/exa-py/";
+    maintainers = with lib.maintainers; [ ethancedwards8 ];
+    license = lib.licenses.mit;
+  };
+})

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5190,6 +5190,8 @@ self: super: with self; {
 
   ewmhlib = callPackage ../development/python-modules/ewmhlib { };
 
+  exa-py = callPackage ../development/python-modules/exa-py { };
+
   example-robot-data = callPackage ../development/python-modules/example-robot-data {
     inherit (pkgs) example-robot-data;
   };

